internal/service/dhcp/ipv6fixedaddress_data_source_test.go

```diff
@@ -0,0 +1,173 @@
+		resource.TestCheckResourceAttrPair(resourceName, "snmp_credential", dataSourceName, "result.0.snmp_credential"),
+		resource.TestCheckResourceAttrPair(resourceName, "template", dataSourceName, "result.0.template"),
+		resource.TestCheckResourceAttrPair(resourceName, "use_cli_credentials", dataSourceName, "result.0.use_cli_credentials"),
+		resource.TestCheckResourceAttrPair(resourceName, "use_domain_name", dataSourceName, "result.0.use_domain_name"),
+		resource.TestCheckResourceAttrPair(resourceName, "use_domain_name_servers", dataSourceName, "result.0.use_domain_name_servers"),
+		resource.TestCheckResourceAttrPair(resourceName, "use_logic_filter_rules", dataSourceName, "result.0.use_logic_filter_rules"),
+		resource.TestCheckResourceAttrPair(resourceName, "use_options", dataSourceName, "result.0.use_options"),
+		resource.TestCheckResourceAttrPair(resourceName, "use_preferred_lifetime", dataSourceName, "result.0.use_preferred_lifetime"),
+		resource.TestCheckResourceAttrPair(resourceName, "use_snmp3_credential", dataSourceName, "result.0.use_snmp3_credential"),
+		resource.TestCheckResourceAttrPair(resourceName, "use_snmp_credential", dataSourceName, "result.0.use_snmp_credential"),
+		resource.TestCheckResourceAttrPair(resourceName, "use_valid_lifetime", dataSourceName, "result.0.use_valid_lifetime"),
+		resource.TestCheckResourceAttrPair(resourceName, "valid_lifetime", dataSourceName, "result.0.valid_lifetime"),
+	}
+}
+
+func testAccIpv6fixedaddressDataSourceConfigFilters(ipv6addr, matchClient, macAddress, networkView, ipv6Network string) string {
+	config := fmt.Sprintf(`
+resource "nios_dhcp_ipv6fixedaddress" "test" {
+ ipv6addr = %q
+ match_client = %q
+ mac_address = %q
+ network = nios_ipam_ipv6network.test_ipv6_network.network
+ network_view = nios_ipam_network_view.parent_network_view.name
+}
+
+data "nios_dhcp_ipv6fixedaddress" "test" {
+ filters = {
+	 ipv6addr = nios_dhcp_ipv6fixedaddress.test.ipv6addr
+ }
+}
+`, ipv6addr, matchClient, macAddress)
+	return strings.Join([]string{testAccBaseNetworkView(networkView, ipv6Network), config}, "")
+}
+
+func testAccIpv6fixedaddressDataSourceConfigExtAttrFilters(ipv6addr, matchClient, macAddress, extAttrsValue, networkView, ipv6Network string) string {
+	config := fmt.Sprintf(`
+resource "nios_dhcp_ipv6fixedaddress" "test" {
+ ipv6addr = %q
+ match_client = %q
+ mac_address = %q
+ network = nios_ipam_ipv6network.test_ipv6_network.network
+ network_view = nios_ipam_network_view.parent_network_view.name
+ extattrs = {
+   Site = %q
+ }
+}
+
+data "nios_dhcp_ipv6fixedaddress" "test" {
+ extattrfilters = {
+	Site = nios_dhcp_ipv6fixedaddress.test.extattrs.Site
+ }
+}
+`, ipv6addr, matchClient, macAddress, extAttrsValue)
+	return strings.Join([]string{testAccBaseNetworkView(networkView, ipv6Network), config}, "")
+}
+
+func testAccBaseNetworkView(networkView, ipv6Network string) string {
```

Comment:
It would be better to rename this function to maintain consistency with other parts of code
```suggestion
func testAccBaseWithNetworkView(networkView, ipv6Network string) string {
```